scipy/interpolate/_pade.py

```diff
@@ -54,8 +54,8 @@ def pade(an, m, n=None):
     if N > len(an)-1:
         raise ValueError("Order of q+p <m+n> must be smaller than len(an).")
     an = an[:N+1]
-    Akj = eye(N+1, n+1)
-    Bkj = zeros((N+1, m), 'd')
+    Akj = eye(N+1, n+1, dtype=an.dtype)
+    Bkj = zeros((N+1, m), dtype=an.dtype)
```

Comment:
If this is something we want to do, unit tests for the enhanced functionality would likely be a good idea.

Reply:
I can do this sometime later today (unless you _really_ want to). I figure there is no reason not to do this - it should work with any previously-working use case, but it will now also work for complex numbers (and probably other use cases that I can't think of).

Reply:
Could you also point me to where these unit tests would go? I haven't contributed to Scipy before, and I want to make sure I put things in the right place.

Reply:
Good question! I didn't know offhand either so I did what I often do with `git grep 'pade('` and found the appropriate file, `scipy/interpolate/tests/test_pade.py`.

If you're feeling really strongly that tests aren't needed you can wait for an interpolate guru to drop in & comment; they may agree with you--I tend to always suggest tests :)

Reply:
+1 to what Tyler said, new functionality needs to be covered by tests. It's not immediately obvious what is going to happen with complex/integer arrays down the line, with `linalg.solve` and `np.poly1d`. And the right way to answer that is to add a test :-).

Here I'd say you need a test with complex inputs, *and* a test with integer arrays/ lists. 

Pade tests are here: https://github.com/scipy/scipy/blob/master/scipy/interpolate/tests/test_pade.py

Reply:
Added some tests based on suggestions from @ev-br. Let me know if these are good enough - happy to change formatting, comments, etc. if anyone wants. I tried to make everything as clean and readable as possible (although the complex input test case is pretty ugly).